### PR TITLE
Use ACL server for CI

### LINKED ISSUE #8128: Remove circular dependency on Fluffy component
## The circle

Following #8121 and other PRs in that sequence, Esti in lakeFS now depends on the external Fluffy component.  Because Fluffy depends on lakeFS, this is something of a circular dependency.  Circular dependencies are bad because they make many otherwise trivial things require careful analysis to unravel.  My general rule to prefer known software engineering techniques by default, and move away from them only as needed.

The customary way to do it would be to test lakeFS using the contrib/ auth server as a mock, and then test Fluffy in Fluffy.

### Example

Here's a concrete example of how this goes bad; there are others.

Suppose we want to add functionality to the auth API.  To be clear: think of a minor-version change, for instance an added optional parameter.  Now an auth server needs to support this feature, and lakeFS will optionally use it.

How do you test and deploy this change?

Before the circular dependency, I would add the parameter to the auth API, implement it on the contrib/ ACL server, and test in Esti on additional tests with this feature switched on.

With the circular dependency things are more complicated:

1. Define, review, and publish the new API -- 1 PR, either on lakeFS or on Fluffy or on both.
1. **Pre-** release Fluffy.  Obviously the new feature on Fluffy cannot yet be tested!
1. Update lakeFS to use the new API feature
1. Update Esti in lakeFS to use the new pre-release Fluffy version.
1. Test lakeFS _and_ Fluffy.  Changes on the client (lakeFS) side are easy to test, but any change to the server (Fluffy) side requires another PR in Fluffy and another pre-release Fluffy version.  That's one PR per attempt to fix a bug -- and often more than one PR per bug!
1. Once everything works, release a new version of Fluffy.
1. Update the lakeFS side PR to use the released version of Fluffy.
1. Push the lakeFS side PR.

So we need at least 3x as many PRs, assuming no bugs.

## Contributor friction

This is smaller, but still unfortunate.

Additionally it removes the ability of contributors to run Esti locally.  The flow for a contributed PR becomes:
1. Contributor writes code on a fork.
1. Contributor can run unit tests locally, but cannot run Esti[^1].
1. Contributor pushes and creates a lakeFS PR.
1. **Core team member** reviews PR.
1. **Core team member** and allows Esti to run.
1. Esti runs for the first time.
So the current changes also add friction to all contributions.

[^1]:  This was possible until now, if you performed some local configuration.

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #8128

## Change Description

Use ACL server for CI instead of fluffy

### CODE REVIEW

**Review by guy-har — APPROVED** [review]
Thanks!

File: esti/ops/docker-compose.yaml
Comment (itaiad200) [inline esti/ops/docker-compose.yaml:1]: Can we rename this file to match the others? `docker-compose-dynamodb`, `docker-compose-external-db` & `docker-compose` is quite confusing.

File: esti/system_test.go [hunk 1]
```diff
@@ -344,6 +345,24 @@ func requireBlockstoreType(t testing.TB, requiredTypes ...string) {
...
 }
 
-func isBasicAuth() bool {
-	return viper.GetBool("auth.basic")
+func isBasicAuth(t testing.TB, ctx context.Context) bool {
+	t.Helper()
+	return getRBACState(t, ctx) == "none"
+}
+
+func isAdvancedAuth(t testing.TB, ctx context.Context) bool {
```
Comment (itaiad200) [inline esti/system_test.go:353]: The ACL server configuration returns `true` for this? If so, it's quite confusing as ACL isn't the advanced auth that we got (it's not RBAC)

**Review by itaiad200 — APPROVED** [review]
Added a few comments that IMO help clarifying some confusing terms. Other than that LGTM

File: .github/workflows/esti.yaml [hunk 2]
```diff
@@ -1361,7 +1421,6 @@ jobs:
           ESTI_AWS_ACCESS_KEY_ID: ${{ secrets.ESTI_AWS_ACCESS_KEY_ID }}
           ESTI_AWS_SECRET_ACCESS_KEY: ${{ secrets.ESTI_AWS_SECRET_ACCESS_KEY }}
           ESTI_VERSION: ${{ needs.deploy-image.outputs.tag }}
-          ESTI_AUTH_BASIC: true
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply .github/workflows/esti.yaml:1364]: Because it is not longer needed. It was used during development

File: esti/ops/docker-compose.yaml
Comment (N-o-Z) [reply esti/ops/docker-compose.yaml:1]: Suggest a name

File: esti/system_test.go (hunk 1 above)
Comment (N-o-Z) [reply esti/system_test.go:353]: No it does not. It return false

**Pull request merged**